packages/zoe/src/zoeService/utils.d.ts

```diff
@@ -102,3 +102,8 @@ export type StartInstance = <SF>(
 export type GetPublicFacet = <SF>(
   instance: Instance<SF> | PromiseLike<Instance<SF>>,
 ) => Promise<StartResult<SF>['publicFacet']>;
+
+export type GetTerms = <SF>(instance: Instance<SF>) => Promise<
+  // only type if 'terms' info is available
+  StartParams<SF>['terms'] extends {} ? StartParams<SF>['terms'] : any
```

Comment:
Conditional types make my non-monotonic-reasoning spidey-sense tingle.

Looking into it a bit, I find others consider them dark magic too:

> Hegel supports ... [the] popular "library definition" format - [TypeScript](http://www.typescriptlang.org/).
> There is only problems with ... conditional types, because Hegel doesn't provide any ... non-type safe concepts for type annotations. -- [Libraries \| Hegel](https://hegel.js.org/docs/libraries)

But the alternative is to bring everything to a screeching halt until all of our contracts have strong types for their terms. So... onward.